### PR TITLE
fix(@spartacus/storefrontlib): Focus lost in quantity input (GH-994)

### PR DESCRIPTION
**Pull request by RadhepS**
Closes: GH-994

### CODE REVIEW

File: projects/storefrontstyles/scss/components/form-components/_item-counter.scss [hunk 1]
```diff
@@ -39,6 +39,11 @@
       color: var(--cx-color, var(--cx-g-color-primary));
     }
 
+    &:focus {
+      color: var(--cx-color, var(--cx-g-color-primary));
+      cursor: pointer;
```
Comment (marlass) [inline projects/storefrontstyles/scss/components/form-components/_item-counter.scss]: I would also increase font-weight when focused. Changing color only is not enough.

File: projects/storefrontlib/src/shared/components/form-components/item-counter/item-counter.component.ts [hunk 2]
```diff
@@ -101,7 +105,6 @@ export class ItemCounterComponent
    * Function set 'isValueOutOfRange' flag and adjust value in range. Then update model value and refresh input
    */
   manualChange(newValue: number): void {
-    this.isValueOutOfRange = this.isOutOfRange(newValue);
```
Comment (marlass) [inline projects/storefrontlib/src/shared/components/form-components/item-counter/item-counter.component.ts]: There is one more mention of this flag. Could you remove it from the comment above `manualChange` function?

File: projects/storefrontlib/src/shared/components/form-components/item-counter/item-counter.component.spec.ts [hunk 3]
```diff
@@ -47,17 +52,43 @@ describe('ItemCounterComponent', () => {
...
+      nativeElement: {
+        value: '',
+        focus: () => {
+          isInputFocused = true;
+        },
+      },
+    };
+    itemCounterComponent.incrementBtn = {
+      nativeElement: {
+        value: '',
```
Comment (marlass) [inline projects/storefrontlib/src/shared/components/form-components/item-counter/item-counter.component.spec.ts]: value is not needed here

File: projects/storefrontlib/src/shared/components/form-components/item-counter/item-counter.component.spec.ts [hunk 4]
```diff
@@ -47,17 +52,43 @@ describe('ItemCounterComponent', () => {
...
+        focus: () => {
+          isIncrementBtnFocused = true;
+        },
+      },
+    };
+    itemCounterComponent.decrementBtn = {
+      nativeElement: {
+        value: '',
```
Comment (marlass) [inline projects/storefrontlib/src/shared/components/form-components/item-counter/item-counter.component.spec.ts]: `value` is not needed here

Comment (RadhepS) [reply projects/storefrontlib/src/shared/components/form-components/item-counter/item-counter.component.spec.ts]: Thank you for the review and changes you suggested. 
I have fixed all the issues you have brought up.

**Pull request merged**